### PR TITLE
feat(ingestion): surface queued document task status

### PR DESCRIPTION
**Pull request by jay77721**
### Summary / Motivation

Expose queued Go ingestion tasks in the dataset document list. With limited ingestion-worker concurrency, a task can be scheduled while the legacy document `run` field is still unstarted, leaving users without feedback and allowing polling to stop after pagination changes.

Fixes #<issue-number>

### Key Changes

- Return the newest Go ingestion task status for each document and report dataset-wide active-task presence.
- Keep document polling active across pages while the dataset contains active ingestion tasks.
- Render queued feedback for `CREATED` and `SCHEDULED` tasks, while keeping cancellation-in-progress internal rather than presenting a separate UI status.
- Centralize processing-state detection for polling and document actions, and prevent conflicting operations while a task is active.
- Add queued translations for all supported frontend locales.
- Add focused DAO, handler, utility, and component test coverage for task-state propagation, polling safety, legacy duplicate rows, and deterministic latest-task selection.

### Implementation Details

- The Go document-list query selects only the latest `ingestion_task` per document, ordered by `create_time` and then `id`, so legacy duplicate rows cannot multiply list results.
- Active-task checks use the same latest-row semantics and a `(dataset_id, status)` index. If the active-task lookup fails, the handler fails open to preserve polling.
- The frontend treats active Go ingestion states as processing in addition to the legacy `run` field. Python responses omit `ingestion_status`, so they retain the existing `run`-based behavior.
- The status-cell cancellation control consistently uses a cancel icon for active tasks; `STOPPING` remains an internal protection state without user-visible status text.

### How to Test

- `bash build.sh --test ./internal/dao -run "^TestIngestionTaskDAOGetByDocumentIDUses(IDAsTieBreaker|LatestTask)$"`
- `npm test -- src/pages/dataset/dataset/parsing-status-cell.test.tsx src/pages/dataset/dataset/utils.test.ts --runInBand --coverage=false`
- In a Go ingestion deployment with a constrained worker pool, submit multiple documents and verify queued documents show queued feedback, dataset polling continues after paging, and active documents cannot be deleted or restarted concurrently.

### CODE REVIEW

**Review by jay77721 — COMMENTED** [review]
## Code Review

Thanks for this PR — the approach is sound and the change is well-scoped. I verified the key claims against `main` and left a few non-blocking suggestions.

### What works well

- The `LEFT JOIN ingestion_task` is safe: `ingestion_task.document_id` carries a unique index (`idx_ingestion_task_document_id`), so the join cannot multiply rows, and `countQuery` (built from `Model(&entity.Document{})` without the join) still produces a correct `total`.
- `isDocumentProcessing` nicely consolidates the four scattered `run === RunningStatus.RUNNING` checks (polling, bulk-delete guard, action cell, status cell), which is a real maintainability win.
- Focused Go (DAO + handler) and frontend (util) tests cover the full propagation path.
- No Python-backend regression: Python's document-list response has no `ingestion_status`, so `undefined === 'SCHEDULED'` is always false and behavior falls back to the previous `run === RUNNING` logic. The shared frontend code stays correct for both backends.

### Suggestions (non-blocking)

**1. Treat all non-terminal ingestion statuses as "processing", not just `SCHEDULED`.** The join exposes the raw task status, which can also be `CREATED`/`RUNNING`/`STOPPING`. If `document.run` ever drifts from the task state (e.g. the `run` update fails or is delayed while the task is already `RUNNING`), `isDocumentProcessing` returns false and the UI stops polling / re-enables delete. The normal path sets `run = "1"` in the same `StartRunning` transition (`internal/service/ingestion_task_service.go:219`), so the window is tiny, but a "non-terminal" check is more robust than a single-value check:

```ts
const ACTIVE = new Set([
  IngestionStatus.CREATED,
  IngestionStatus.SCHEDULED,
  IngestionStatus.RUNNING,
  IngestionStatus.STOPPING,
]);
export const isDocumentProcessing = (d) =>
  isParserRunning(d.run) || ACTIVE.has(d.ingestion_status);
```

**2. `IngestionStatus` enum is incomplete.** Only `SCHEDULED` is declared while the backend can emit `CREATED`/`RUNNING`/`COMPLETED`/`FAILED`/`STOPPED`/`STOPPING`. Adding the full set documents the contract (and is needed for suggestion 1).

**3. Dead `omitempty` tag.** `DocumentListItem.IngestionStatus` has `json:"ingestion_status,omitempty"`, but the handler maps the field manually via `stringValue`, which emits `""` for nil documents — so the field is always present and `omitempty` never applies. Consider dropping `omitempty` to avoid a misleading tag.

**4. i18n completeness.** `runningStatusScheduled` is only added to `en.ts`/`zh.ts`. The other locales (ar/bg/de/es/fr/id/it/ja/ko/pt-br/ru/tr/vi/zh-traditional) will render the raw key. Worth confirming the fallback is acceptable or backfilling the key.

### Minor observation

There are now two parallel "is this document being processed" signals (`run` vs `ingestion_status`). The PR reconciles them at the UI via `isDocumentProcessing`, but the invariant is implicit. A short comment on `IDocumentInfo`/`utils.ts` clarifying the relationship would help future call sites avoid bypassing the helper.

File: internal/dao/document.go [hunk 1]
```diff
@@ -137,11 +137,12 @@ func (dao *DocumentDAO) ListByKBIDWithOptions(ctx context.Context, db *gorm.DB,
...
 
 	listQuery := db.WithContext(ctx).Table("document").
-		Select(`document.*, user_canvas.title as pipeline_name, user.nickname`).
+		Select(`document.*, user_canvas.title as pipeline_name, user.nickname, ingestion_task.status as ingestion_status`).
 		Joins("JOIN file2document ON file2document.document_id = document.id").
 		Joins("JOIN file ON file.id = file2document.file_id").
 		Joins("LEFT JOIN user_canvas ON document.pipeline_id = user_canvas.id").
-		Joins("LEFT JOIN user ON document.created_by = user.id")
+		Joins("LEFT JOIN user ON document.created_by = user.id").
+		Joins("LEFT JOIN ingestion_task ON ingestion_task.document_id = document.id")
```
[the remark this answers is not in the document]
Comment (jay77721) [reply internal/dao/document.go]: Resolved in 5e48cd3. The document-list join now keeps only the newest ingestion_task row per document, with create_time and ID as deterministic ordering. This also protects databases where the migration cannot add the unique index because legacy duplicate rows already exist; the focused document-list DAO test covers that case.

File: internal/dao/ingestion_task_test.go [hunk 2]
```diff
@@ -326,3 +326,40 @@ func TestIngestionTaskDAODeleteIfTerminal_RemovesOnlyTerminal(t *testing.T) {
...
 	}
 }
+
+func TestIngestionTaskDAOCountActiveByDatasetID(t *testing.T) {
+	db := setupTaskTestDB(t)
+	for _, task := range []*entity.IngestionTask{
+		{ID: "task-active-1", UserID: "user-1", DocumentID: "doc-active-1", DatasetID: "kb-1", Status: common.SCHEDULED},
+		{ID: "task-active-2", UserID: "user-1", DocumentID: "doc-active-2", DatasetID: "kb-1", Status: common.RUNNING},
+		{ID: "task-terminal", UserID: "user-1", DocumentID: "doc-terminal", DatasetID: "kb-1", Status: common.COMPLETED},
+		{ID: "task-other-kb", UserID: "user-1", DocumentID: "doc-other", DatasetID: "kb-2", Status: common.SCHEDULED},
```
[the remark this answers is not in the document]
Comment (jay77721) [reply internal/dao/ingestion_task_test.go:338]: Resolved in 6373a8c. The test data now includes CREATED and STOPPING alongside SCHEDULED and RUNNING, and expects all four active tasks.

File: internal/handler/document.go [hunk 3]
```diff
@@ -614,7 +616,11 @@ func (h *DocumentHandler) ListDocuments(c *gin.Context) {
 		docs = append(docs, mapDocumentListItem(doc, metaFields))
 	}
 
-	common.SuccessWithData(c, gin.H{"total": total, "docs": docs}, "success")
+	hasActiveTasks, err := h.documentService.HasActiveIngestionTasks(ctx, datasetID)
+	if err != nil {
+		common.Warn("failed to check active ingestion tasks", zap.Error(err))
+	}
+	common.SuccessWithData(c, gin.H{"total": total, "docs": docs, "has_active_tasks": hasActiveTasks}, "success")
```
[the remark this answers is not in the document]
Comment (jay77721) [reply internal/handler/document.go:626]: Resolved in ac46d14. The handler now fails open by returning has_active_tasks=true when the lookup fails, and the handler test asserts that polling remains active in this error path.

File: internal/entity/ingestion_task.go [hunk 4]
```diff
@@ -20,9 +20,9 @@ type IngestionTask struct {
 	ID         string  `gorm:"column:id;primaryKey;size:32" json:"id"`
 	UserID     string  `gorm:"column:user_id;size:32;not null" json:"user_id"`
 	DocumentID string  `gorm:"column:document_id;size:32;not null;uniqueIndex:idx_ingestion_task_document_id" json:"document_id"`
```
[the remark this answers is not in the document]
Comment (jay77721) [reply internal/entity/ingestion_task.go:22]: Not adopting this schema change. The current Go ingestion model intentionally enforces one task per document: Create rejects duplicates and the unique index is the concurrency backstop. Selecting the newest row is defensive handling for legacy duplicate rows when the migration cannot add that index, not a new task-history contract. Allowing multiple normal task rows would change retry and cancellation lifecycle semantics beyond this PR.

**Comment by jay77721** [comment]
Review follow-up:

- @jay77721 suggestion 1 and 2: resolved in 6373a8c. isDocumentProcessing treats CREATED, SCHEDULED, RUNNING, and STOPPING as active, and the frontend enum declares the full backend contract.
- @jay77721 suggestion 3: not changing omitempty. The current handler conditionally adds ingestion_status only when it is present; it does not use the stringValue behavior described in the review. The tag has no endpoint effect and removing it would only alter direct struct serialization.
- @jay77721 suggestion 4: resolved in 6373a8c. The queued key was backfilled to the remaining locales.
- Minor observation: resolved in 6373a8c with the utils.ts contract comment, including the Python fallback behavior.
- CodeRabbit tie-breaker nitpick: resolved in b876042fb. A focused DAO test now uses equal create_time values and reverse insertion order, then asserts that the higher ID is selected.
- The frontend intentionally does not present STOPPING as a user-visible state. It remains an internal active/protection state, while the UI shows queued feedback only for CREATED/SCHEDULED.
